### DISCUSSION TITLE
`/ollama/api/generate`  api for customGPTs

### DISCUSSION THREAD

**Discussion by veya2ztn** [open-webui/open-webui#4061] [General]
The openwebup support api usage like `[OWEB]/ollama/api/generate` for base mode loaded from ollama. 
However, if I want to wrapper the base model (for example, the `qwen2` model) with certern 
- system prompt
- knowledge base for RAG
The `[OWEB]/ollama/api/generate` will return ` status code: 400`.

below is a sample that get property result via  `[OWEB]/ollama/api/generate` , the model named `qwen2:72b-instruct-q4_0`
![image](https://github.com/open-webui/open-webui/assets/18117225/b53e2760-797c-42ab-9539-788f65b5c4de)

If I wrapper it in a customGPTs that even with no any extra setting. (we name it `qwentest`)
![image](https://github.com/open-webui/open-webui/assets/18117225/c0ff7353-9201-41d6-ac82-2ba5171ac4ee)
----------
# The reason for such suggestion.
The `[OWEB]/ollama/api/generate` is usually for the `code completion` task. Currently, it barely base on the ability of the base model such as starcode2 or deepseek-code. Enable preset prompt may help to build much more robust code. Moreover, I also wise the `code completion`  can eventully become `text completion` that for `writting as copilot way`

**Comment by justinh-rahb** [comment]
> Maybe you should try it：
> [link redacted]

GTFO @812781385
